cmd/jsonnet.cpp

```diff
@@ -673,38 +685,76 @@ int main(int argc, const char **argv)
 
             case FMT: {
                 std::string output_file = config.outputFile;
-                if (config.fmtInPlace) {
-                    if (config.inputFile == "-") {
-                        std::cerr << "ERROR: Cannot use --in-place with stdin" << std::endl;
-                        jsonnet_destroy(vm);
-                        return EXIT_FAILURE;
+
+                if (config.fmtInPlace || config.fmtTest) {
+                    assert(config.inputFiles.size() >= 1);
+                    for (std::string inputFile: config.inputFiles) {
+                        if (config.fmtInPlace) {
+                            output_file = inputFile;
+
+                            if (inputFile == "-") {
+                                std::cerr << "ERROR: Cannot use --in-place with stdin" << std::endl;
+                                jsonnet_destroy(vm);
+                                return EXIT_FAILURE;
+                            }
+                            if (config.filenameIsCode) {
+                                std::cerr << "ERROR: Cannot use --in-place with --exec" << std::endl;
+                                jsonnet_destroy(vm);
+                                return EXIT_FAILURE;
+                            }
+                        }
+
+                        std::string input;
+                        if (!read_input_contents(inputFile, &input)) {
+                            jsonnet_destroy(vm);
+                            return EXIT_FAILURE;
+                        }
+
+                        output = jsonnet_fmt_snippet(vm, inputFile.c_str(), input.c_str(), &error);
+
+                        if (error) {
```

Comment:
I see two ways to approach this:
a) Process everything as we do know, but keep the results in memory and write all of them at once afterwards.
b) Just do the initial validation on all files (check that they exist, that stdin is not used with -i, maybe that they can be opened in case permissions are wrong).

Of course, even with a), it's still possible that only some of them will be updated if something bad happens (file removed, permissions changed, process killed).